python/ray/ml/utils/remote_storage.py

```diff
@@ -0,0 +1,154 @@
+import re
+from typing import Optional, Tuple
+
+try:
+    import fsspec
+except ImportError:
+    fsspec = None
+
+try:
+    import pyarrow
+    import pyarrow.fs
+except (ImportError, ModuleNotFoundError):
+    pyarrow = None
+
+from ray import logger
+
+URI_PROTOCOL_REGEX = r"^(\w+)://(.+)$"
+S3_PREFIX = "s3://"
+GS_PREFIX = "gs://"
+HDFS_PREFIX = "hdfs://"
+ALLOWED_REMOTE_PREFIXES = (S3_PREFIX, GS_PREFIX, HDFS_PREFIX)
```

Comment:
Are these considered "legacy"?
Should we comment on what is the new expected uri schemes? Like "s3fs" instead of "s3"?